src/Context.php

```diff
@@ -237,25 +188,30 @@ public function notice($type, $message, \PhpParser\NodeAbstract $expr, $status =
      */
     public function syntaxError(\PhpParser\Error $exception, $filepath)
     {
-        $code = file($filepath);
+        return $this->addIssue('syntax-error', $exception->getMessage(), $exception->getStartLine(), $filepath);
+    }
+
+    /**
+     * @param $type
+     * @param $message
+     * @param $line
+     * @param string|null $filepath
+     * @return bool
+     */
+    protected function addIssue($type, $message, $line, $filepath = null)
```

Comment:
IssueLocation in near feature will be different and will support blocks
and half of line